tests/unit/common/core/test_middleware.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from pytest_mock import MockerFixture
+
+from common.core import middleware as middleware_module
+from common.core import utils
+
+
+@pytest.fixture(autouse=True)
+def clear_lru_caches() -> None:
+    utils.get_file_contents.cache_clear()
+    utils.get_versions_from_manifest.cache_clear()
```

Comment:
I guess we'd need to consider here that, if we take my suggestion above to make this function private, that we'd need to access a private function to clear the lru cache. I guess there's no reason that we couldn't keep the function public though tbh.

Reply:
See https://github.com/Flagsmith/flagsmith-common/pull/59#discussion_r2058866844